src/dlload.c

```diff
@@ -60,18 +60,18 @@ extern char *julia_bindir;
 
 #define JL_RTLD(flags, FLAG) (flags & JL_RTLD_ ## FLAG ? RTLD_ ## FLAG : 0)
 
-static void JL_NORETURN jl_dlerror(const char *fmt, const char *sym)
+static const char * jl_dlerror(void)
 {
 #ifdef _OS_WINDOWS_
-    CHAR reason[256];
+    static CHAR reason[256];
```

Comment:
This makes this no longer thread safe, which is probably fine if we only do it in the error case, but if we do it in the success case as well, I'm worried the error message will get overwritten by the success of a different thread.

Reply:
Unfortunately, we can't assume that that's universally available.

Reply:
If you really need to, you can use our own tls... There're already platform dependent members in there so it shouldn't be a huge problem even though it won't look as nice and will certainly needs cross references in the comments.

Reply:
Because there seems to be some disagreement within the Julia codebase, I have taken the liberty of adding `#define JL_THREAD_LOCAL` to `julia.h` which emulates what is held within `src/support/dirname.c`.  If this shouldn't be done (and we should instead use the more complex TLS implementation in `src/threading.c`), it's probably best to transition both `jl_dlerror()` and `dirname()` to use that new stuff instead.